src/databricks/labs/remorph/reconcile/compare.py

```diff
@@ -106,6 +115,85 @@ def reconcile_data(
     )
 
 
+def reconcile_agg_data(
+    source: DataFrame,
+    target: DataFrame,
+    group_list: list[Aggregate],
+) -> DataReconcileOutput:
+    source_alias = "src"
+    target_alias = "tgt"
+
+    select_columns = [
+        (f"source_{agg.type}_{agg_col}".lower(), f"target_{agg.type}_{agg_col}".lower())
+        for agg in group_list
+        for agg_col in agg.agg_cols
+    ]
+
+    group_columns = None
+    df = source.alias(source_alias).join(
+        other=target.alias(target_alias),
+        how="cross",
+    )
+
+    if group_list[0].group_by_cols:
+        group_columns = [
+            (f"source_group_by_{group_col}", f"target_group_by_{group_col}")
+            for group_col in group_list[0].group_by_cols
+        ]
+
+        select_columns = select_columns + group_columns
+        df = source.alias(source_alias).join(
+            other=target.alias(target_alias),
+            on=_generate_agg_join_condition(source_alias, target_alias, group_columns),
+            how="full",
+        )
+
+    df = df.selectExpr(
+        *source.columns,
+        *target.columns,
+    ).cache()
+
+    df.show()
+
+    # Write unmatched df to volume
+    # volume_df = ReconIntermediatePersist(spark, path + f"/{group_key}").write_and_read_unmatched_df_with_volumes(df)
+    # logger.warning(f"Unmatched data is written to {path} successfully")
+
+    # volume_df.show()
+
+    mismatch = _get_mismatch_agg_data(df, select_columns, group_columns)
+
+    mismatch.show()
+
+    missing_in_src = df.filter(_agg_conditions(select_columns, "missing_in_src")).select(*target.columns)
+    missing_in_src.show()
+
+    missing_in_tgt = df.filter(_agg_conditions(select_columns, "missing_in_tgt")).select(*source.columns)
+    missing_in_tgt.show()
+
+    mismatch_count = 0
+    if mismatch:
+        mismatch_count = mismatch.count()
+
+    missing_in_src_count = missing_in_src.count()
+    missing_in_tgt_count = missing_in_tgt.count()
+
+    print(
+        f"mismatch_count: {mismatch_count}",
+        f"missing_in_src_count: {missing_in_src_count}",
+        f"missing_in_tgt_count: {missing_in_tgt_count}",
+    )
+
+    return DataReconcileOutput(
+        mismatch_count=mismatch_count,
+        missing_in_src_count=missing_in_src_count,
+        missing_in_tgt_count=missing_in_tgt_count,
+        missing_in_src=missing_in_src.limit(_SAMPLE_ROWS),
+        missing_in_tgt=missing_in_tgt.limit(_SAMPLE_ROWS),
+        mismatch=MismatchOutput(mismatch_df=mismatch),
+    )
+
+
 def _get_mismatch_data(df: DataFrame, src_alias: str, tgt_alias: str) -> DataFrame:
```

Comment:
We can reuse by moving key column as parm to the function and reduce the introduction of `_get_mismatch_agg_data` function